docs/docs/en/user-guide/runtimes/autogen/index.md

```diff
@@ -1,132 +1,189 @@
 ```
 
 #### 5. **Register API Functions with the Agents**
-In this step, we register the weather API functions to ensure that the weather agent can call the correct functions, such as `get_daily_weather` and `get_daily_weather_weekly_get`, to retrieve the required weather data.
+In this step, we register the [**weather API**](https://weather.tools.fastagency.ai/docs){target="_blank"} functions to ensure that the weather agent can call the correct functions, such as `get_daily_weather` and `get_daily_weather_weekly_get`, to retrieve the required weather data.
 
 ```python
-{! docs_src/user_guide/runtimes/autogen/main.py [ln:52-65] !}
+{! docs_src/user_guide/runtimes/autogen/mesop/main.py [ln:58-72] !}
 ```
 
 #### 6. **Enable Agent Interaction and Chat**
-Here, the user agent initiates a chat with the weather agent, which queries the API and returns the weather information. The conversation is summarized using a method provided by the LLM.
+Here, the user agent initiates a chat with the weather agent, which queries the [**weather API**](https://weather.tools.fastagency.ai/docs){target="_blank"} and returns the weather information. The conversation is summarized using a method provided by the [**LLM**](https://en.wikipedia.org/wiki/Large_language_model){target="_blank"}.
 
 ```python
-{! docs_src/user_guide/runtimes/autogen/main.py [ln:67-74] !}
+{! docs_src/user_guide/runtimes/autogen/mesop/main.py [ln:73-81] !}
 ```
 
 #### 7. **Create and Run the Application**
-Finally, we create the FastAgency application and launch it using the console interface.
+Finally, we create the FastAgency application and launch it using the [**`mesop`**](../../../api/fastagency/ui/mesop/MesopUI.md) interface.
 
 ```python
-{! docs_src/user_guide/runtimes/autogen/main.py [ln:77] !}
+{! docs_src/user_guide/runtimes/autogen/mesop/main.py [ln:83] !}
 ```
 
 ### Complete Application Code
 
 <details>
 <summary>main.py</summary>
 ```python
-{! docs_src/user_guide/runtimes/autogen/main.py!}
+{! docs_src/user_guide/runtimes/autogen/mesop/main.py!}
 ```
 </details>
 
 
 ### Running the Application
 
-```bash
-fastagency run
-```
+!!! note
 
-Ensure you have set your OpenAI API key in the environment and that the weather API URL is accessible. The command will launch a console interface where users can input their requests and interact with the weather agent.
+    In this example, AutoGen will utilize its [**code execution**](https://microsoft.github.io/autogen/0.2/docs/tutorial/code-executors#use-code-execution-in-conversation){target="_blank"} feature to execute external [**Rest APIs**](https://en.wikipedia.org/wiki/REST){target="_blank"} and retrieve their responses. For security reasons, AutoGen defaults to running code execution inside a [**Docker**](https://www.docker.com/){target="_blank"} container. If Docker is not running on the machine where you are executing this example, you will encounter the following error:
```

Comment:
This is not true. AutoGen will use tool calling to access remote REST API, it will not generate and execute code.

Reply:
Removed this section from docs. It is not relevant anymore.